Dfe.Academies.Academisation.Seed/SeedProject.cs

```diff
@@ -0,0 +1,85 @@
+﻿using AutoFixture;
+using Dfe.Academies.Academisation.Data;
+using Dfe.Academies.Academisation.Data.ProjectAggregate;
+using Dfe.Academies.Academisation.Domain.Core.ProjectAggregate;
+using Dfe.Academies.Academisation.Domain.ProjectAggregate;
+using FakerDotNet;
+
+namespace Dfe.Academies.Academisation.Seed;
+
+public static class SeedProject
+{
+	public static async Task CreateProject(AcademisationContext academisationContext, int numberOfProjects)
+	{
+		var fixture = new Fixture();
+		var newProjectStates = new List<ProjectState>();
+		var projectNotes = new List<ProjectNoteState>();
+		for (int i = 1; i <= numberOfProjects; i++)
+		{
+			var newAcademyConversionProject = NewAcademyConversionProject();
+			using var dbContextTransaction = await academisationContext.Database.BeginTransactionAsync();
+			try
+			{
+				// Create and add project
+				var newProjectState = ProjectState.MapFromDomain(newAcademyConversionProject);
+				newProjectState.Id = default;
+				newProjectStates.Add(newProjectState);
+
+				// Batch save for every 80 project (If over 80)
+				if (newProjectStates.Count % 80 == 0 || numberOfProjects < 80)
+				{
+					academisationContext.Projects.AddRange(newProjectStates);
```

Comment:
I was hoping EF offered something more... sophisticated than this. A quick search shows that maybe it did _once upon a time_ but it's all abandoned now :(